app/controllers/albums.js

```diff
@@ -11,3 +17,32 @@ exports.getAlbumPhotos = (req, res, next) =>
   listAlbumPhotos(req.params.id)
     .then(response => res.status(200).send(response.body))
     .catch(next);
+
+exports.buyAlbum = (req, res, next) =>
+  getAlbumData(req.params.id)
+    .then(albumData =>
+      findAlbum(albumData.id, req.user.id).then(existingAlbum => {
```

Comment:
Do you know about the method `findOrCreate` ? I think that It would be nice if you use it